tests/mltoolbox_magic/shell_process_tests.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+"""Test the datalab prediction with MLToolbox models."""
+from __future__ import absolute_import
+from __future__ import print_function
+
+import logging
+import os
+import psutil
+import six
+import subprocess
+import sys
+import threading
+import time
+import unittest
+
+import google.datalab.contrib.mltoolbox._shell_process as _shell_process
+
+
+class TestShellProcess(unittest.TestCase):
+  """Tests for process managements used in MLToolbox magics."""
+
+  def setUp(self):
+    self._logger = logging.getLogger('TestStructuredDataLogger')
+    self._logger.setLevel(logging.DEBUG)
+    if not self._logger.handlers:
+      self._logger.addHandler(logging.StreamHandler(stream=sys.stdout))
+
+  def test_process(self):
```

Comment:
This test depends on events happening within time intervals. I would like to see a comment/documentation outlining the timeline of events . Something like

step 1: process_to_wait is created with 10 seconds to live. It is assumed this test will finish before then.
step 2: monitor process is created, which then makes a worker process. This test sleeps for 1 sec
step 3: worker process prints "exclude message"
step 4: worker sleeps for 1 sec
step 5: worker prints include message 1
step 6: This test wakes up (after the 1 sec), kills process_to_wait
step 7: monitor process seeks process_to_wait is dead, and kills the worker